bot/helper/reqHelper.ts

```diff
@@ -0,0 +1,202 @@
+        const lines = segmentListSharp[1];
+        const lineNumList = lines.replace(/L/g,"").split("-");
+        // Get startLine and endLine
+        startLine = Number(lineNumList[0]) -1;
+        endLine = Number(lineNumList[1]) -1;  
+        fileurl = segmentListSharp[0]
+    }
+    const segmentList = fileurl.replace("https://github.com/","").replace("blob/","").split("/");
+    // name space is the first element of the segmentList.
+    const namespace = segmentList.shift();
+    // repo name is the second element of the segmentList.
+    const repoName = segmentList.shift();
+    // ref is the third part of the segmentList, ref means the name of the commit/branch/tag.
+    const ref = segmentList.shift();
+    // connect the remaining part to get file path.
+    const path = segmentList.reduce((a,b) => a + "/" + b);
+   
+    var content: string;
+    // Use third-part API to render code to HTML.
+    if (!useGithubRendered){
+        // Request whole content from GitHub API with metadata.
+        const rawContent = await reqInfoFromGitHubAPI(namespace, repoName, path, ref);
+        const contentToRender = segmentContent(rawContent, startLine, endLine);
+        const language = getLanguage(path);
+        content = await renderContent(language, contentToRender);
+    }
+    // Directly use GitHub-rendered HTML of the file.
+    // Notice: No syntax-highlighting; Hard to get certain lines of code from rendered HTML.
+    else {
+        content = await reqRenderedHTMLFromGitHub(namespace, repoName, path);
+    }
+
+    var card = new CodeCard(`[${ref}]${namespace}/${repoName}/${path}: Lines ${startLine+1} - ${endLine+1}`, content, url);
+    return card;
+}
+
+/**
+ * Function to request file data from GitHub API.
+ * @param namespace namespace
+ * @param repoName name of the repository
+ * @param path path of the file
+ * @param ref name of the commit/branch/tag
+ * @returns raw content string of the file
+ */
+ async function reqInfoFromGitHubAPI(namespace: string, repoName: string, path: string, ref: string){
+    if (ref == undefined){
+        ref = 'main'
+    }
+    const reqURL = `https://api.github.com/repos/${namespace}/${repoName}/contents/${path}`;
+    var content:string;
+    await axios({
+        baseURL: reqURL,
+        method: 'get',
+        headers: {
+            'Accept':'application/vnd.github+json'
+        },
+        params:{
+            'ref': ref
+        }
+    }).then( (response) => {
+        // Need to decode with base64
+        content = Buffer.from(response.data["content"],'base64').toString('utf-8');    
+    });
+    return content;
+}
+
+/**
+ * Function to get certain lines of a file content.
+ * @param content 
+ * @param startLine the index of the starting line
+ * @param endLine the index of the ending line
+ * @returns content string of certain lines of a file content
+ */
+ function segmentContent(content:string, startLine:number, endLine: number){
```

Comment:
Not quite understand for this function, let me check with you offline.

Reply:
From the API we could get the whole file content, using this to intercept the select lines